### PR TITLE
Install PMIx v3.1.5

### PR DESCRIPTION
**Pull request by tilne**
This enables the slurm be built against it and use it when launching
jobs.

Avoid issues like [this one](https://github.com/aws/aws-parallelcluster/issues/1182) by passing `--mpi=pmix` to `srun`.

Signed-off-by: Tim Lane <tilne@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by tilne** [comment]
Updating this with two changes:
* missed adding `libevent-devel` to the packages list for centos6
* the apt package equivalent is named `libevent-dev` instead of `libevent-devel`

Also rebasing on develop

File: attributes/default.rb [hunk 1]
```diff
@@ -161,7 +165,7 @@
                                                 httpd boost-devel redhat-lsb mlocate mpich-devel openmpi-devel R atlas-devel
                                                 blas-devel fftw-devel libffi-devel openssl-devel dkms mysql-devel libedit-devel
                                                 libical-devel postgresql-devel postgresql-server sendmail mdadm python python-pip
-                                                libgcrypt-devel]
+                                                libgcrypt-devel libevent-devel]
```
Comment (demartinofra) [inline attributes/default.rb:168]: isn't this getting applied to centos 6 too?

Comment (tilne) [reply attributes/default.rb:168]: Yes. This is the package list for centos6 right?

Comment (demartinofra) [reply attributes/default.rb:168]: sorry just me misreading your update :D

**Pull request merged**